Makefile

```diff
@@ -228,6 +228,7 @@ docker-telemetrygen:
 .PHONY: generate
 generate:
 	cd cmd/mdatagen && $(GOCMD) install .
+	$(GOCMD) install github.com/Khan/genqlient@v0.6.0
```

Comment:
Opted for adding this line to install geqlient & then the corresponding `//go:generate ` snippet in the github_scraper.go file

Reply:
That was the original route I tried, based on perusing the repository and the way other tools were installed. However, when installing through tools, `go generate` isn't able to pick up the path to the tool, because the `.tools` directory, which is the location all the tools are installed, isn't part of the path. 

All the other tools commands in the makefile [create an argument with the full path](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/Makefile.Common#L57-L70) to the command after install in the .tools directory. 

This doesn't work though for `go generate` which is what I suspect was the reason that mdatagen was chosen to be installed in the `cd cmd/mdatagen && $(GOCMD) install .` option.

Reading through the [wiki linked in tools.go](https://github.com/golang/go/wiki/Modules#how-can-i-track-tool-dependencies-for-a-module) it looks like tools.go was the preferred path until the `go install path@version` paradigm presented in go 1.16+ 

To be fair, I like the ability to define the version via tools because it enabled package management through the native go functionality. However, I think given the scenario and intent behind make generate, it makes the most sense to be installed here. 

The other option I see is that we could put it in tools, and add a specific section in the gitprovider makefile or the common one to add the argument for generate with a new target. I do prefer the simplicity of the go generate option via make generate though.

Reply:
I am not sure about how we should manage moving forward. I got this going and it seems to work ok: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/29298

The main issue of installing genqlient like this is that you won't get updates from dependabot.

Reply:
Of course I missed the simple option on leveraging relative paths for the command 😆 It's always the simple / little things. 

In the workflows they run `make install-tools` prior to running all the make commands. I'm totally good with the target being called by generate as well, but just wanted to mention it in case you feel the duplication of running install-tools in the workflow would add unnecessary time. Locally it seems to add only a few seconds but just wanted to call it out. 

Thanks for putting up a PR of a better way to make that work!